HL7800 PPP support

rerickson1 opened the pull request:
Add support for using CMUX and PPP with the Pinnacle 100 modem (HL7800).

sample: net: cellular_modem now works with the Pinnacle 100 DVK and MG100 gateway.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/65836

rerickson1 committed:
boards: mg100: add hl7800 node label

Add HL7800 modem node label.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

rerickson1 committed:
boards: pinnacle_100_dvk: add hl7800 node label

Add HL7800 modem node label.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

rerickson1 committed:
samples: net: cellular_modem: fix kconfig warning for logging

Logging needs to be on for CONFIG_MODEM_MODULES_LOG_LEVEL_DBG to work.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

rerickson1 committed:
samples: net: cellular_modem: add pinnacle 100 support

Add support for Pinnacle 100 DVK.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

rerickson1 committed:
samples: net: cellular_modem: add mg100 support

Add support for mg100 gateway.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

bjarki-andreasen reviewed (dismissed):
Really nice to see it finally working!

rerickson1 committed:
modem: modem_cellular: fixes for hl7800

Add connect match for dialing PPP connection.
Ensure PDP authentication is disabled.
Request manufacturer, firmware version, and IMSI from the modem.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
